Recorder/OutputType.h

```diff
@@ -258,11 +259,16 @@ enum class OutputType {
 
     // amplitude
     AMP,
-    NL
+    NL = 256
 };
 
-const char* to_name(OutputType);
-OutputType to_token(const char*);
+template<> struct magic_enum::customize::enum_range<OutputType> {
```

Comment:
If you need some help, I can show you an example if you add me as collaborator in your repo settings: https://github.com/lokitkhemka/suanPan/settings/access

Reply:
I will keep that in mind. I have also given you access to my fork of the project. Thank you so much. My most common way of using git is me individually working on my own project. This is my first time contributing to someone else's project. It was not planned. I just reading the code for my own learning (I love FEA and wanted to learn how to implement it and this project was a gold mine, especially with constitutive models, particularly related to Geotech because that is my specialization, academically speaking), and when I found I can actually contributed, I just did. I will spend some time getting better at collaboration.

Reply:
It's all fine, it's a learning process for everyone.

Cool if you are interested in write something new in geotech.
I always wanted to have some practical examples of what have been implemented but just had no chance to do so.
Maybe, if you are willing to, you can also try to write some examples and contribute to the documentation repo, it will be much appreciated!

Reply:
Just FYI, maybe you have already seen it, all implementation details of (most) constitutive models are documented in this book: https://github.com/TLCFEM/constitutive-modelling-cookbook

Reply:
Yes. I had the pdf. Also, I really want to add some geotech related examples, but can you tell me how you convert a mesh into a format that can be used as input in suanpan, especially for the more complicated meshes? I mean, for example, I create a mesh in gmsh or any other software, is there a way to automate it to convert it into a format that readable by suanpan. From ,y understanding, In Suanpan, there is only a list of nodes in one file and elements in another.

I am just throwing some ideas so that it can easy to implement newer examples.

Reply:
gmsh allows you to export the mesh to a list of coordinates (for nodes) and a list of element connectivity (node order), for example, export the mesh as vtk file.
Other formats may be even easier to work with, but I do not know much of the details.

What it stores is very similar to what suanPan uses.
Then it is possible to further convert it by some python scripting.

gmsh also has the concept of set, which can be converted to groups (node groups or element groups).

Reply:
Awesome. I will look to see if I can create some examples. Slope analysis is a prototypical geotechnical problem, which is where I intended to start.